### DISCUSSION TITLE
Keep Modal Open

### DISCUSSION THREAD

**Discussion by korematter** [filamentphp/filament#8060] [Help]
Is it possible to keep a modal open after a livewire server request for form validation? 

I have a modal form that I dispatch on-click.

```
<form wire:submit.prevent="saveDancer(true,false)">
    <x-filament::modal
        id="custom-add-leader-modal"
        width="2xl">

        <x-slot name="header">
            <div class="font-bold">
                Add New Leader
            </div>
        </x-slot>

       -- form fields --

        <x-slot name="actions">
            <div class="flex space-x-4">
            <x-filament::button x-on:click="isOpen=false" color="secondary" class="w-1/2">
                Cancel
            </x-filament::button>
            <x-filament::button type="submit" color="primary" class="w-1/2">
                Save
            </x-filament::button>
            </div>
        </x-slot>

    </x-filament::modal>
</form>
```

I have validation that should be throwing validationException but modal keeps closing.

```
public function saveDancer($is_leader, $is_follower)
    {
        // validation here
    }
```

Is there a way to keep the modal open?

Thanks,
Steve

**Comment by leandrocfe** [comment]
Are you using form builder? https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component
Why not validate with `$this->form->getState()`?

**Reply by korematter** [reply]
Thanks @leandrocfe . Unfortunately no. This is a custom Page with unique field requirements so I'm launching the modal via alpine dispatch. I just need it to stay open when making livewire server requests for validation. Not sure why the modal would be closing.

**Reply by leandrocfe** [reply]
maybe `type="button" wire:click="yourMethod()"`
But I would use Form builder..

**Reply by korematter** [reply]
Thank you leandrocfe. Same behavior. Will have another look at Form Builder but will be a complete refactor. The browser dispatch method has been recommended on other discussions so it must be possible to do form validations on a modal without having the modal closing. Will keep at it. Thanks again.
